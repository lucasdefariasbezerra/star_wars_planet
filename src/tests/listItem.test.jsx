import React from 'react';
import ListItem from '../components/shared/listItem';
import { shallow } from 'enzyme';

let wrapper;
const props = {
    item: { title: 'test', director: 'lewis test', producer: 'lewis producer' }
};

describe('</ListItem>', () => {
    beforeEach(() => {
        wrapper = shallow(
            <ListItem { ...props} />
        );
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should match props', () => {
        const title = wrapper.dive().find('FeedLabel').at(1).props().content;
        const feedChildren = wrapper.find('FeedSummary').props().children;
        const summaryText = feedChildren.reduce((text, item) => text.concat(item.concat(' ')));
        expect(title).toEqual('test');
        expect(summaryText.trim()).toEqual('director: lewis test  producer:  lewis producer');
    });


});