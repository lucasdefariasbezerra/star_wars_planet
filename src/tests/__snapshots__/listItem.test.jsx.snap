// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</ListItem> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    item={
      Object {
        "director": "lewis test",
        "producer": "lewis producer",
        "title": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FeedLabel
          image="https://cdn2.iconfinder.com/data/icons/space-stuff-1/68/bb8-512.png"
        />,
        <FeedContent>
          <FeedLabel
            content="test"
          />
          <FeedSummary>
            director: 
            lewis test
             producer: 
            lewis producer
          </FeedSummary>
        </FeedContent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "image": "https://cdn2.iconfinder.com/data/icons/space-stuff-1/68/bb8-512.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <FeedLabel
              content="test"
            />,
            <FeedSummary>
              director: 
              lewis test
               producer: 
              lewis producer
            </FeedSummary>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "content": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "director: ",
                "lewis test",
                " producer: ",
                "lewis producer",
              ],
            },
            "ref": null,
            "rendered": Array [
              "director: ",
              "lewis test",
              " producer: ",
              "lewis producer",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FeedLabel
            image="https://cdn2.iconfinder.com/data/icons/space-stuff-1/68/bb8-512.png"
          />,
          <FeedContent>
            <FeedLabel
              content="test"
            />
            <FeedSummary>
              director: 
              lewis test
               producer: 
              lewis producer
            </FeedSummary>
          </FeedContent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "image": "https://cdn2.iconfinder.com/data/icons/space-stuff-1/68/bb8-512.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <FeedLabel
                content="test"
              />,
              <FeedSummary>
                director: 
                lewis test
                 producer: 
                lewis producer
              </FeedSummary>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "content": "test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "director: ",
                  "lewis test",
                  " producer: ",
                  "lewis producer",
                ],
              },
              "ref": null,
              "rendered": Array [
                "director: ",
                "lewis test",
                " producer: ",
                "lewis producer",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
